octobot_trading/exchanges/exchange_manager.py

```diff
@@ -109,42 +115,51 @@ def reset_exchange_personal_data(self):
         self.exchange_type = RestExchange.create_exchange_type(self.exchange_class_string)
 
         if not self.is_backtesting:
-            # create REST based on ccxt exchange
-            self.exchange = RestExchange(self.config, self.exchange_type, self)
-            await self.exchange.initialize()
-
-            self._load_constants()
-
-            if not self.exchange_only:
-                await self._create_exchange_channels()
-
-            # create Websocket exchange if possible
-            if not self.rest_only:
-                # search for websocket
-                if self.check_web_socket_config(self.exchange.name):
-                    await self._search_and_create_websocket(AbstractWebsocket)
-
-        # if simulated : create exchange simulator instance
+            # real : create a rest or websocket exchange instance
+            await self._create_real_exchange()
         else:
-            self.exchange = ExchangeSimulator(self.config, self.exchange_type, self, self.backtesting_files)
-            await self.exchange.initialize()
-            self.exchange_config.set_config_traded_pairs()
-            await self._create_exchange_channels()
+            # simulated : create exchange simulator instance
+            await self._create_simulated_exchange()
 
         if not self.exchange_only:
             # create exchange producers if necessary
             await self._create_exchange_producers()
 
         if self.is_backtesting:
-            try:
-                await self.exchange.modify_channels()
-                await self.exchange.create_backtesting_exchange_producers()
-            except ValueError:
-                self.logger.error("Not enough exchange data to calculate backtesting duration")
-                await self.stop()
+            await self._init_simulated_exchange()
 
         self.is_ready = True
 
+    async def _create_real_exchange(self):
+        # create REST based on ccxt exchange
+        self.exchange = RestExchange(self.config, self.exchange_type, self)
+        await self.exchange.initialize()
+
+        self._load_constants()
+
+        if not self.exchange_only:
+            await self._create_exchange_channels()
+
+        # create Websocket exchange if possible
+        if not self.rest_only:
+            # search for websocket
+            if self.check_web_socket_config(self.exchange.name):
+                await self._search_and_create_websocket(AbstractWebsocket)
+
+    async def _create_simulated_exchange(self):
+        self.exchange = ExchangeSimulator(self.config, self.exchange_type, self, self.backtesting_files)
+        await self.exchange.initialize()
+        self.exchange_config.set_config_traded_pairs()
+        await self._create_exchange_channels()
+
+    async def _init_simulated_exchange(self):
```

Comment:
No need for this and the 3 previous methods in pxd ?

Reply:
It's not possible to declare async methods in pxd.